harvest/api/dummy.py

```diff
@@ -92,45 +92,83 @@ def fetch_price_history(self,
+            end = pytz.utc.localize(end)
+
+        # Convert datetime to indices
+        start_index = int((start - epoch_zero()).total_seconds() // 60)
+        end_index = 1 + int((end - epoch_zero()).total_seconds() // 60)
+
+        if symbol in self.randomness:
+            # If we already generated data from this asset
+
+            num_of_random = 1 + end_index
+
+            if len(self.randomness[symbol]) < num_of_random:
+                # If the new end index is greater than the data we have
+
+                # Get the rng for this symbol
+                rng = self.randomness[symbol + '_rng'] 
+
+                # Generate a bunch of random numbers from -0.5 to 0.5
+                increments = rng.random(num_of_random - len(self.randomness[symbol])) - 0.499
+
+                # Get sum of the previous price changes
+                latest_price_change = self.randomness[symbol][-1]
+
+                # Calculate the change in price since the first price
+                new_price_changes = latest_price_change + np.cumsum(increments) 
+
+                # Store the new prices
+                self.randomness[symbol] = np.append(self.randomness[symbol], new_price_changes)
+            
         else:
-            error(f'Interval unit was {unit} not MIN, HR, or DAY!')
+            # If there is no information about the asset 
+
+            # Create an rng using the asset's symbol as a seed
+            rng = np.random.default_rng(int.from_bytes(symbol.encode('ascii'), 'big'))
+            num_of_random = 1 + end_index
+            increments = rng.random(num_of_random) - 0.499
+
+            # Store the price change since the first price
+            self.randomness[symbol] = np.cumsum(increments)
+            self.randomness[symbol + '_rng'] = rng
+
+        # The inital price is arbitarly calculated from the first change in price
+        start_price = 100 * (self.randomness[symbol][0] + 0.51)     
 
         times = []
-        current = start
-
-        stock_gen = self._generate_fake_stock_data()
-        open_s = []
-        high = []
-        low = []
-        close = []
-        volume = []
-
-        # Fake the data 
-        while current < end + interval:
-            times.append(current)
-            current += interval
-
-            o, h, l, c, v = next(stock_gen)
-            open_s.append(o)
-            high.append(h)
-            low.append(l)
-            close.append(c)
-            volume.append(v)         
+        current_time = start
+
+        # Get the prices for the current interval
+        prices = start_price + self.randomness[symbol][start_index:end_index]
+        # Prevent prices from going negative
+        prices[prices < 0] = 0.01
+
+        # Calculate olhcv from the prices
+        open_s = 0.95 * prices
+        low = 0.8 * prices 
+        high = 1.2 * prices 
+        close = 1.05 * prices 
+        volume = (1000 * (prices + 20)).astype(int)
```

Comment:
Currently the only source of randomness is the price, in the future we can increase the number of random values to create more variance in the data.